Protocol.md

```diff
@@ -296,6 +296,76 @@ The recipient:
 
   These script pub key forms include only standard forms accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners.
 
+# Message segmentation
+
+The communication protocol specified in [BOLT #8](https://github.com/lightningnetwork/lightning-rfc/blob/master/08-transport.md) is based on the noise protocol that has a fixed limit of 65535 bytes for message lengths.
+This can be insufficient for contract with large number of outcomes (and thus adaptor signatures).
+To circumvent this issue, a simple message segmentation protocol is defined, that splits large messages into a `segment_start` message followed by 
+
+## The `segment_start` message
+
+The segment start message indicates the start of a message that is split in several segments.
+
+1. type: 42900 (`segment_start`)
+1. data:
+  * [`u16`:`nb_segments`]
+  * [`bigsize`:`data_len`]
+  * [`data_len*u8`:`data`]
+
+### Requirements
+
+The sender MUST:
+  - set `nb_segments` to the number of segments into which the message will be split into
+  - set `nb_segments` to a number greater than 1 and smaller than 1000
```

Comment:
Since you have a fixed payload size (`data_len`) you could also send the overall payload size (reassembled buffer length from all segments) and the number of segments could be inferred from that. Knowing the size ahead of time precisely could allow the receiver to allocate a buffer that exactly fits, instead of having to either overallocate or expand for each segment.

Reply:
Then again, having the `data_len` in there I expect you might want to allow for other `data_len` sizes in future, making the number of segments unknown, but then we can use a greedy strategy of just consuming segments while the buffer hasn't been filled.

Reply:
You made two very good points, so if IIUC you're saying either:
* have a fixed number of segments, using a single `data_len` and `nb_segments` in the start message
* have a variable number of segments, using a `data_len` per message (I guess it still requires a `start` though)

I can't think of a case where we would need actual streaming of data, but leaving the possibility open is tempting, tough choice...